packages/interfaces/package.json

```diff
@@ -62,5 +62,8 @@
     "CHANGELOG.md",
     "LICENSE",
     "README.md"
-  ]
+  ],
+  "dependencies": {
+    "@waku/proto": "^0.0.6"
```

Comment:
I don't think we should add it as `dependency` to `@waku/interfaces`

Reply:
why is that? 🤔

Reply:
`@waku/interface` name implies to be `interfaces` only and to have little to no runtime footprint. as I mentioned somewhere in your PRs - right now we abuse `enums` and use them as `const` (note: this is not necessarily bad thing with `enums`)

as for `@waku/proto` - it has enormous runtime footprint and has nothing to do with `ts` types directly 

we should think of work around here:
- defining type in `@waku/interfaces` and using in `@waku/proto` instead
- duplicating type in `@waku/interfaces` and using it from here instead of `@waku/proto` 

maybe you think of something else?

Reply:
> defining type in @waku/interfaces and using in @waku/proto instead

this seems challenging because `proto` ts files are generated from `.proto` definitions, and we shouldn't be manually updating those

> duplicating type in @waku/interfaces and using it from here instead of @waku/proto

this is what we have been doing so far

Reply:
> this is what we have been doing so far

I probably missed it. Could you, please, point me where? 

I checked build artifact and it doesn't contain any footprint of `proto`, that means it is smart enough to understand there is only type import

as a precaution I propose to make it `devDependency` (whole `@waku/interfaces` is kinda a dev dependnecy anyway) and change `import { proto_store as proto } from "@waku/proto";` to something like `import type { proto_store as proto } from "@waku/proto";` in `store.ts` 

This way we ensure it is not increasing size and it doesn't leak anything else apart from types

Reply:
> I probably missed it. Could you, please, point me where?

For example: `Cursor`:
`@waku/proto` definition: https://github.com/waku-org/js-waku/blob/0a8382e7a293b0b3b6d94b29f194696ae1372c03/packages/proto/src/lib/store.ts#L11-L16
`@waku/interfaces` redefinition: https://github.com/waku-org/js-waku/blob/0a8382e7a293b0b3b6d94b29f194696ae1372c03/packages/interfaces/src/store.ts#L14-L19

> 
> as a precaution I propose to make it devDependency (whole @waku/interfaces is kinda a dev dependnecy anyway) and change import { proto_store as proto } from "@waku/proto"; to something like import type { proto_store as proto } from "@waku/proto"; in store.ts
> 
> This way we ensure it is not increasing size and it doesn't leak anything else apart from types

I agree with this, approach wise